seems that `lint` is failing due to internal issues?

```
make[2]: Entering directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/oteltestbedcol'
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
make[2]: *** [../../Makefile.Common:166: lint] Error 4
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/configschema'
make[1]: *** [Makefile:167: cmd/configschema] Error 2
make[1]: *** Waiting for unfinished jobs....
running /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/misspell -error
Check License finished successfully
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/golangci-lint run --allow-parallel-runners --build-tags integration
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/oteltestbedcol'
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make: *** [Makefile:105: golint] Error 2
```

--- a/.chloggen/drosiek-host-id.yaml
+++ b/.chloggen/drosiek-host-id.yaml
@@ -1,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: resourcedetectionprocessor
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add host.id to system detector
+
+# One or more tracking issues related to the change
+issues: [18533]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:

--- a/cmd/configschema/go.mod
+++ b/cmd/configschema/go.mod
@@ -188,6 +188,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsperfcountersreceiver v0.0.0-00010101000000-000000000000 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver v0.0.0-00010101000000-000000000000 // indirect
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zookeeperreceiver v0.0.0-00010101000000-000000000000 // indirect
+	github.com/panta/machineid v1.0.2 // indirect
 	github.com/signalfx/signalfx-agent/pkg/apm v0.0.0-20230214151822-6a6813cf5bf1 // indirect
 )
 

--- a/cmd/otelcontribcol/go.mod
+++ b/cmd/otelcontribcol/go.mod
@@ -495,6 +495,7 @@ require (
 	github.com/openzipkin/zipkin-go v0.4.1 // indirect
 	github.com/outcaste-io/ristretto v0.2.0 // indirect
 	github.com/ovh/go-ovh v1.3.0 // indirect
+	github.com/panta/machineid v1.0.2 // indirect
 	github.com/patrickmn/go-cache v2.1.0+incompatible // indirect
 	github.com/paulmach/orb v0.8.0 // indirect
 	github.com/pelletier/go-toml v1.9.5 // indirect

--- a/exporter/datadogexporter/go.mod
+++ b/exporter/datadogexporter/go.mod
@@ -136,6 +136,7 @@ require (
 	github.com/openshift/api v0.0.0-20210521075222-e273a339932a // indirect
 	github.com/openshift/client-go v0.0.0-20210521082421-73d9475a9142 // indirect
 	github.com/outcaste-io/ristretto v0.2.0 // indirect
+	github.com/panta/machineid v1.0.2 // indirect
 	github.com/philhofer/fwd v1.1.2 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect

--- a/go.mod
+++ b/go.mod
@@ -511,6 +511,7 @@ require (
 	github.com/openzipkin/zipkin-go v0.4.1 // indirect
 	github.com/outcaste-io/ristretto v0.2.0 // indirect
 	github.com/ovh/go-ovh v1.3.0 // indirect
+	github.com/panta/machineid v1.0.2 // indirect
 	github.com/patrickmn/go-cache v2.1.0+incompatible // indirect
 	github.com/paulmach/orb v0.8.0 // indirect
 	github.com/philhofer/fwd v1.1.2 // indirect

--- a/internal/metadataproviders/go.mod
+++ b/internal/metadataproviders/go.mod
@@ -7,6 +7,7 @@ require (
 	github.com/aws/aws-sdk-go v1.44.201
 	github.com/docker/docker v23.0.1+incompatible
 	github.com/hashicorp/consul/api v1.18.0
+	github.com/panta/machineid v1.0.2
 	github.com/stretchr/testify v1.8.1
 )
 

--- a/internal/metadataproviders/system/metadata.go
+++ b/internal/metadataproviders/system/metadata.go
@@ -22,6 +22,7 @@ import (
 	"strings"
 
 	"github.com/Showmax/go-fqdn"
+	"github.com/panta/machineid"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/internal"
 )
@@ -61,6 +62,9 @@ type Provider interface {
 
 	// ReverseLookupHost does a reverse DNS query on the current host's IP address
 	ReverseLookupHost() (string, error)
+
+	// HostID returns Host Unique Identifier
+	HostID() (string, error)
 }
 
 type systemMetadataProvider struct {
@@ -122,4 +126,8 @@ func (p systemMetadataProvider) reverseLookup(ipAddresses []string) (string, err
 		return strings.TrimRight(names[0], "."), nil
 	}
 	return "", fmt.Errorf("reverseLookup failed to convert IP addresses to name: %w", err)
+}
+
+func (p systemMetadataProvider) HostID() (string, error) {
+	return machineid.ID()
 }

--- a/processor/resourcedetectionprocessor/README.md
+++ b/processor/resourcedetectionprocessor/README.md
@@ -35,6 +35,7 @@ Note: use the Docker detector (see below) if running the Collector as a Docker c
 Queries the host machine to retrieve the following resource attributes:
 
     * host.name
+    * host.id
     * os.type
 
 By default `host.name` is being set to FQDN if possible, and a hostname provided by OS used as fallback.

--- a/processor/resourcedetectionprocessor/go.mod
+++ b/processor/resourcedetectionprocessor/go.mod
@@ -72,6 +72,7 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/common v0.71.0 // indirect
 	github.com/opencontainers/go-digest v1.0.0 // indirect
 	github.com/opencontainers/image-spec v1.0.2 // indirect
+	github.com/panta/machineid v1.0.2 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/rogpeppe/go-internal v1.9.0 // indirect

--- a/processor/resourcedetectionprocessor/internal/system/system.go
+++ b/processor/resourcedetectionprocessor/internal/system/system.go
@@ -70,12 +70,19 @@ func (d *Detector) Detect(_ context.Context) (resource pcommon.Resource, schemaU
 	if err != nil {
 		return res, "", fmt.Errorf("failed getting OS type: %w", err)
 	}
+
+	hostID, err := d.provider.HostID()
+	if err != nil {
+		return res, "", fmt.Errorf("failed getting host ID: %w", err)
+	}
+
 	for _, source := range d.hostnameSources {
 		getHostFromSource := hostnameSourcesMap[source]
 		hostname, err = getHostFromSource(d)
 		if err == nil {
 			attrs.PutStr(conventions.AttributeHostName, hostname)
 			attrs.PutStr(conventions.AttributeOSType, osType)
+			attrs.PutStr(conventions.AttributeHostID, hostID)
 
 			return res, conventions.SchemaURL, nil
 		}

--- a/processor/resourcedetectionprocessor/internal/system/system_test.go
+++ b/processor/resourcedetectionprocessor/internal/system/system_test.go
@@ -51,6 +51,11 @@ func (m *mockMetadata) OSType() (string, error) {
 	return args.String(0), args.Error(1)
 }
 
+func (m *mockMetadata) HostID() (string, error) {
+	args := m.MethodCalled("HostID")
+	return args.String(0), args.Error(1)
+}
+
 func (m *mockMetadata) LookupCNAME() (string, error) {
 	args := m.MethodCalled("LookupCNAME")
 	return args.String(0), args.Error(1)
@@ -92,6 +97,7 @@ func TestDetectFQDNAvailable(t *testing.T) {
 	md := &mockMetadata{}
 	md.On("FQDN").Return("fqdn", nil)
 	md.On("OSType").Return("darwin", nil)
+	md.On("HostID").Return("2", nil)
 
 	detector := &Detector{provider: md, logger: zap.NewNop(), hostnameSources: []string{"dns"}}
 	res, schemaURL, err := detector.Detect(context.Background())
@@ -102,6 +108,7 @@ func TestDetectFQDNAvailable(t *testing.T) {
 	expected := map[string]any{
 		conventions.AttributeHostName: "fqdn",
 		conventions.AttributeOSType:   "darwin",
+		conventions.AttributeHostID:   "2",
 	}
 
 	assert.Equal(t, expected, res.Attributes().AsRaw())
@@ -113,6 +120,8 @@ func TestFallbackHostname(t *testing.T) {
 	mdHostname.On("Hostname").Return("hostname", nil)
 	mdHostname.On("FQDN").Return("", errors.New("err"))
 	mdHostname.On("OSType").Return("darwin", nil)
+	mdHostname.On("OSType").Return("darwin", nil)
+	mdHostname.On("HostID").Return("3", nil)
 
 	detector := &Detector{provider: mdHostname, logger: zap.NewNop(), hostnameSources: []string{"dns", "os"}}
 	res, schemaURL, err := detector.Detect(context.Background())
@@ -123,6 +132,7 @@ func TestFallbackHostname(t *testing.T) {
 	expected := map[string]any{
 		conventions.AttributeHostName: "hostname",
 		conventions.AttributeOSType:   "darwin",
+		conventions.AttributeHostID:   "3",
 	}
 
 	assert.Equal(t, expected, res.Attributes().AsRaw())
@@ -132,6 +142,7 @@ func TestUseHostname(t *testing.T) {
 	mdHostname := &mockMetadata{}
 	mdHostname.On("Hostname").Return("hostname", nil)
 	mdHostname.On("OSType").Return("darwin", nil)
+	mdHostname.On("HostID").Return("1", nil)
 
 	detector := &Detector{provider: mdHostname, logger: zap.NewNop(), hostnameSources: []string{"os"}}
 	res, schemaURL, err := detector.Detect(context.Background())
@@ -142,6 +153,7 @@ func TestUseHostname(t *testing.T) {
 	expected := map[string]any{
 		conventions.AttributeHostName: "hostname",
 		conventions.AttributeOSType:   "darwin",
+		conventions.AttributeHostID:   "1",
 	}
 
 	assert.Equal(t, expected, res.Attributes().AsRaw())
@@ -153,6 +165,7 @@ func TestDetectError(t *testing.T) {
 	mdFQDN.On("OSType").Return("windows", nil)
 	mdFQDN.On("FQDN").Return("", errors.New("err"))
 	mdFQDN.On("Hostname").Return("", errors.New("err"))
+	mdFQDN.On("HostID").Return("", errors.New("err"))
 
 	detector := &Detector{provider: mdFQDN, logger: zap.NewNop(), hostnameSources: []string{"dns"}}
 	res, schemaURL, err := detector.Detect(context.Background())
@@ -164,6 +177,7 @@ func TestDetectError(t *testing.T) {
 	mdHostname := &mockMetadata{}
 	mdHostname.On("OSType").Return("windows", nil)
 	mdHostname.On("Hostname").Return("", errors.New("err"))
+	mdHostname.On("HostID").Return("", errors.New("err"))
 
 	detector = &Detector{provider: mdHostname, logger: zap.NewNop(), hostnameSources: []string{"os"}}
 	res, schemaURL, err = detector.Detect(context.Background())
@@ -175,6 +189,7 @@ func TestDetectError(t *testing.T) {
 	mdOSType := &mockMetadata{}
 	mdOSType.On("FQDN").Return("fqdn", nil)
 	mdOSType.On("OSType").Return("", errors.New("err"))
+	mdOSType.On("HostID").Return("", errors.New("err"))
 
 	detector = &Detector{provider: mdOSType, logger: zap.NewNop(), hostnameSources: []string{"dns"}}
 	res, schemaURL, err = detector.Detect(context.Background())

